samples/drivers/CAN/src/main.c

```diff
@@ -208,73 +136,52 @@ void main(void)
 	can_configure(can_dev, CAN_LOOPBACK_MODE, 250000);
 #endif
 
-	led_gpio_dev = device_get_binding(CONFIG_GPIO_LED_DEV);
+#if defined(DT_ALIAS_LED0_GPIOS_PIN) && defined(DT_ALIAS_LED0_GPIOS_CONTROLLER)
+	led_gpio_dev = device_get_binding(DT_ALIAS_LED0_GPIOS_CONTROLLER);
 	if (!led_gpio_dev) {
 		printk("LED: Device driver not found.\n");
 		return;
 	}
 
-	k_sem_init(&tx_sem, 0, INT_MAX);
-
-	button_gpio_dev = device_get_binding(CONFIG_GPIO_BUTTON_DEV);
-	if (!button_gpio_dev) {
-		printk("Button: Device driver not found.\n");
-		return;
-	}
-
-	ret = gpio_pin_configure(button_gpio_dev, CONFIG_PIN_USER_BUTTON,
-				    (GPIO_DIR_IN | GPIO_INT | GPIO_INT_EDGE |
-				     GPIO_INT_ACTIVE_HIGH | GPIO_INT_DEBOUNCE));
-	if (ret) {
-		printk("Error configuring  button pin\n");
-	}
-
-	gpio_init_callback(&gpio_cb, button_callback,
-			   BIT(CONFIG_PIN_USER_BUTTON));
-
-	ret = gpio_add_callback(button_gpio_dev, &gpio_cb);
-	if (ret) {
-		printk("Cannot setup callback!\n");
-	}
-
-	ret = gpio_pin_enable_callback(button_gpio_dev, CONFIG_PIN_USER_BUTTON);
-	if (ret) {
-		printk("Error enabling callback!\n");
+	ret = gpio_pin_configure(led_gpio_dev, DT_ALIAS_LED0_GPIOS_PIN,
+				 GPIO_DIR_OUT);
+	if (ret < 0) {
+		printk("Error setting LED pin to output mode [%d]", ret);
 	}
+#endif
 
-	ret = can_attach_isr(can_dev, rx_button_isr, 0, &filter);
+	ret = can_attach_workq(can_dev, &k_sys_work_q, &rx_work, change_led,
+			       led_gpio_dev, &change_led_filter);
 	if (ret == CAN_NO_FREE_FILTER) {
 		printk("Error, no filter available!\n");
 		return;
 	}
 
-	k_tid_t tx_tid = k_thread_create(&tx_thread_data, tx_thread_stack,
-					 K_THREAD_STACK_SIZEOF(tx_thread_stack),
-					 tx_thread,
+	printk("Change LED filter ID: %d\n", ret);
+
+	k_tid_t rx_tid = k_thread_create(&rx_thread_data, rx_thread_stack,
```

Comment:
Variable definitions at the beginning of the functions.

Reply:
Thanks.